File: Python/ExampleTests/types_glue.py
from common import *
from production import Dollar, SimpleText


class TypesGlue:
    DNC_STRING = '?DNC?'

    def examples_DataType_Dollar(self, values: list):
        for value in values:
            vvt = ValidValuesTyped.from_string_obj(value)
            failed = False
            try:
                Dollar(vvt.value)
            except ValueError:
                failed = True
            assert vvt.is_valid == (not failed), f' Value {vvt.value}'

    def examples_DataType_SimpleText(self, values: list):
        for value in values:
            vvt = ValidValuesTyped.from_string_obj(value)
            failed = False
            try:
                SimpleText(vvt.value)
            except ValueError:
                failed = True
            assert vvt.is_valid == (not failed), f' Value {vvt.value}'

File: JavaScript/ExampleTests/types_glue.js
import { ValidValuesTyped } from "./common/index.js";
import { Dollar, SimpleText } from "./production/index.js";

export class TypesGlue {
  static DNC_STRING = "?DNC?";

  examplesDataTypeDollar(values) {
    values.forEach((value) => {
      const vvt = ValidValuesTyped.fromStringObj(value);
      let failed = false;
      try { new Dollar(vvt.value); } catch (e) { failed = true; }
      expect(vvt.isValid).toBe(!failed);
    });
  }

  examplesDataTypeSimpleText(values) {
    values.forEach((value) => {
      const vvt = ValidValuesTyped.fromStringObj(value);
      let failed = false;
      try { new SimpleText(vvt.value); } catch (e) { failed = true; }
      expect(vvt.isValid).toBe(!failed);
    });
  }
}
